From 9ba93658e616a250a10cb8956e2e5db0fa1cbd0f Mon Sep 17 00:00:00 2001
From: Sami Shalayel <sami.shalayel@qt.io>
Date: Tue, 23 Sep 2025 11:57:46 +0200
Subject: [PATCH 098/239] QQmlJSUtils::resourceFilesFromBuildFolders: port to
 QDirListing

Use QDirListing instead QDirIterator in resourceFilesFromBuildFolders.
QDirIterator is deprecated and shows up a bit in the profiler when
investigating why tst_qmlls_modules::qmldirImports timeouts some time on
a debug asan build.

Change-Id: Icc80c39f48ac381828829dc0d4c5eb51ed42abbf
Reviewed-by: Fabian Kosmale <fabian.kosmale@qt.io>
(cherry picked from commit 9e70404a6398a661dc386c51c12b173000fd80fa)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/qmlcompiler/qqmljsutils.cpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/qmlcompiler/qqmljsutils.cpp b/src/qmlcompiler/qqmljsutils.cpp
index f725ba67ba..09a13d4792 100644
--- a/src/qmlcompiler/qqmljsutils.cpp
+++ b/src/qmlcompiler/qqmljsutils.cpp
@@ -253,10 +253,11 @@ QStringList QQmlJSUtils::resourceFilesFromBuildFolders(const QStringList &buildF
 {
     QStringList result;
     for (const QString &path : buildFolders) {
-        QDirIterator it(path, QStringList{ u"*.qrc"_s }, QDir::Files | QDir::Hidden,
-                        QDirIterator::Subdirectories);
-        while (it.hasNext()) {
-            result.append(it.next());
+        for (auto it : QDirListing{ path, QStringList{ u"*.qrc"_s },
+                                    QDirListing::IteratorFlag::Recursive
+                                            | QDirListing::IteratorFlag::FilesOnly
+                                            | QDirListing::IteratorFlag::IncludeHidden }) {
+            result.append(it.filePath());
         }
     }
     return result;
-- 
2.51.2

